client/my-sites/checkout/checkout/secure-payment-form.jsx

```diff
@@ -259,11 +251,19 @@ export class SecurePaymentForm extends Component {
 		}
 	}
 
-	finishIfLastStep( cart, selectedSite, step ) {
+	async finishIfLastStep( cart, selectedSite, step ) {
 		if ( ! step.last || step.error ) {
 			return;
 		}
 
+		try {
+			await this.maybeSetSiteToPublic( { cart } );
+		} catch ( e ) {
+			debug( 'Error setting site to public', e );
+			displayError();
+			return;
```

Comment:
This early return and error message could be quite worrying to the user. The error message will say `There was a problem completing the checkout. Please try again.` but the payment has been successful. And I don't think they'll be able to try again (and they really shouldn't!)